#110 - filter for locally missing items as well

lmihalkovic opened the pull request:
an implementation for #110

insidegui commented:
I can't verify this right now because I'm on vacation but as soon as I get back I'll take a look. Thanks

lmihalkovic commented:
I realized the downloaded/missing is reversed... :(  I fixed it in my local repo and can push another PR if you'd like... in the meantime, this is what the change entails. Apologies for the inconvenience.

```
Precedes: 

    FIX download filter

------------------------ WWDC/FilterBarController.swift ------------------------
index 01fa32a..aab9213 100644
@@ -105,7 +105,7 @@ class FilterBarController: NSViewController, GRScrollViewDelegate {
             focusMenu.addItem(item)
         }
         
-        for (name,state) in [("Downloaded","yes"),("Missing","no")] {
+        for (name,state) in [("Downloaded","true"),("Missing","false")] {
             let item = NSMenuItem(title: "\(name)", action: "downloadedMenuAction:", keyEquivalent: "")
             item.state = downloadedFilter.selectedStrings!.contains(state) ? NSOnState : NSOffState
             item.target = self

------------------------------ WWDC/Models.swift ------------------------------
index 6d8d46e..18f3747 100644
@@ -146,7 +146,7 @@ enum SearchFilter {
         case .Favorited(let favorited):
             return NSPredicate(format: "favorite = %@", favorited)
         case .Downloaded(let downloaded):
-            return NSPredicate(format: "downloaded = %@", downloaded[0].boolValue)
+            return NSPredicate(format: "downloaded = %@", !downloaded[0].boolValue)
         }
     }
```     

PS: I implemented grouping by year. I can create a branch for that if you see any value in it (I like to track the evolution). I see no reason why it could not be extended to group by other fields (I pass a closure to extract the criteria).

insidegui commented:
Yes please open another PR to fix this :)

And please do the grouping and open a PR, I think It's a nice feature to add ;)
